backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/reader/spi/IndexReaderHolder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.lowlevel.reader.spi;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import org.apache.lucene.index.IndexReader;
+
+public interface IndexReaderHolder extends Closeable {
```

Comment:
Currently it's not very useful. At some point we'll have to restore several advanced strategies from Search 5. Some of these strategies create index readers differently, for example one strategy periodically creates a single index reader for all threads to share, and keeps a reference counter, only discarding the index reader when there reference counter is 0. This SPI allows implementing that kind of advanced resource management, since `IndexReader#close()` is no longer called directly.